src/main/scala/analysis/data_structure_analysis/Graph.scala

```diff
@@ -189,6 +189,21 @@ class Graph(val proc: Procedure,
     global
   }
 
+  // determine if an address is a global and return the corresponding global(s) if it is.
+  def getGlobal(address: BigInt, size: Int): Seq[DSAGlobal] =
+    var global: Seq[DSAGlobal] = Seq.empty
+    breakable {
+      for (elem <- globalMapping) {
+        val range = elem._1
+        val field = elem._2
```

Comment:
You can just do `for ((range, field) <- globalMapping)` instead, which is simpler and clearer